File: api/src/domain/entities/Movimiento.ts
export type TipoMovimiento = 'entrada' | 'salida';

export interface Movimiento {
  id?: number;
  tipo: TipoMovimiento;
  fecha: Date;
  cantidad: number;
  motivo: string;
  insumo_id: number;
  bodega_id: number;
  created_at?: Date;
}

File: web/core/domain/movimiento.py
from dataclasses import dataclass
from datetime import date

@dataclass
class Movimiento:
    id: int
    tipo: str
    fecha: date
    cantidad: int
    motivo: str
    insumo_id: int
    bodega_id: int
    insumo_nombre: str = ""
    bodega_nombre: str = ""